### ISSUE TITLE
Tried to archive a users page that has over 27K tweets but I ended up getting  200 tweets.

### ISSUE THREAD

**Issue by DiameterEffect** [Benjins/Anarchivist#3]
Is this a error? I dont get all the persons tweets, images, and videos. 
![image](https://user-images.githubusercontent.com/65675954/152711624-705dc041-a10c-4fa8-bf7b-8d720c6fe98c.png)

**Comment by Benjins** [comment]
So this is a bug (I can see the user has more tweets than the ones being captured), but I don't know the immediate reason. The way the script works is equivalent to searching "from:username" in the Twitter search bar. However, searching that for this user doesn't give all the tweets they've made; other users seem to still work okay.

You can try again at a later date and see if maybe it's just a blip in Twitter's backend. I'll try poking at it a bit more to see if there's some reason for it. If it's important, you can also try to find a scraper that uses the Twitter API which shouldn't have this sort of issue.

I was going to recommend trying snscrape, but it seems to be running into the same issue where not all tweets are able to be fetched. I believe it uses the same method of searching for "from:username" so this makes sense

**Comment by Benjins** [comment]
It appears to be another case of https://github.com/JustAnotherArchivist/snscrape/issues/4 

Not sure what causes it, possibly something like a shadow ban. May resolve itself, but may not

**Comment by DiameterEffect** [comment]
> It appears to be another case of [JustAnotherArchivist/snscrape#4](https://github.com/JustAnotherArchivist/snscrape/issues/4)
> 
> Not sure what causes it, possibly something like a shadow ban. May resolve itself, but may not

Odd, im going to try it again right now and see if it gives me the same issue.

**Comment by DiameterEffect** [comment]
> It appears to be another case of [JustAnotherArchivist/snscrape#4](https://github.com/JustAnotherArchivist/snscrape/issues/4)
> 
> Not sure what causes it, possibly something like a shadow ban. May resolve itself, but may not

okay Ive tried it with another user account and it sorta works but it stops and doesnt collect all of the 17k tweets.
![image](https://user-images.githubusercontent.com/65675954/152896646-f58b6a6b-659e-4c50-a156-28604ab6bddd.png)

**Comment by Benjins** [comment]
Which account is that? Are there tweets that don't get scraped by the script, but show up in a "from:username" search?

**Comment by DiameterEffect** [comment]
from the account called FLCLSHOTS 
> Which account is that? Are there tweets that don't get scraped by the script, but show up in a "from:username" search?

Yes, it looks like i downloaded everything from the "from:username" search, but is that really all of the 17K tweet that user has?

**Comment by Benjins** [comment]
Yeah, the script only gets ~2.8K tweets from that account, whereas snscrape gets all 14.7K. I'll see if there's some backend change I need to adapt to

**Comment by Benjins** [comment]
Okay, I added a retry mechanism in 5b6d3e12c94825a7248c7f555310beca95e5fd9e that seems to get around this issue: the search query was failing several times on some of the cursors at around the June 2021 mark for that user

I'd like to figure out if there's a better way, but things should be working: you'll see some retry messages in the logs. I only saw it fail ~15 times for that user, so I set the max retries to 20. If that still doesn't work you can try tweaking it in the script, or there's some deeper bug

**Comment by DiameterEffect** [comment]
> Okay, I added a retry mechanism in [5b6d3e1](https://github.com/Benjins/Anarchivist/commit/5b6d3e12c94825a7248c7f555310beca95e5fd9e) that seems to get around this issue: the search query was failing several times on some of the cursors at around the June 2021 mark for that user
> 
> I'd like to figure out if there's a better way, but things should be working: you'll see some retry messages in the logs. I only saw it fail ~15 times for that user, so I set the max retries to 20. If that still doesn't work you can try tweaking it in the script, or there's some deeper bug

okay thank you i'll try this later. The thing about snscrape, I used it before and it just doesnt work like this one. Im not sure how to get media,images, and tweets with snscrape since it requires python knowledge which I dont have yet.
